inst/include/mode/r/mode.hpp

```diff
@@ -24,10 +26,25 @@ double mode_time(SEXP ptr) {
   return obj->time();
 }
 
+cpp11::sexp state_array(const std::vector<double>& dat,
+                        size_t n_state,
+                        size_t n_particles) {
+  cpp11::writable::doubles ret(dat.size());
+  std::copy(dat.begin(), dat.end(), REAL(ret));
+
+  ret.attr("dim") = cpp11::writable::integers{n_state, n_particles};
+
+  return ret;
+}
+
 template <typename T>
-std::vector<double> mode_solve(SEXP ptr, int end_time) {
+cpp11::sexp mode_solve(SEXP ptr, int end_time) {
   T *obj = cpp11::as_cpp<cpp11::external_pointer<T>>(ptr).get();
```

Comment:
We can throw anywhere really, but we might as well do it here as then we can have nicer string interpolation.

```
if (end_time < obj->time()) {
  cpp11::stop("end_time (%f) must be greater than current time (%f)", end_time, obj->time());
}
```

otherwise in the current implementation if you run the model to some time (say 25) then solve to time 1 it returns y at time 25